Library/Homebrew/mktemp.rb

```diff
@@ -35,7 +35,7 @@ def to_s
   end
 
   def run
-    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix}-", HOMEBREW_TEMP))
+    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix.dup.tr! "@", "-"}-", HOMEBREW_TEMP))
```

Comment:
```suggestion
    @tmpdir = Pathname.new(Dir.mktmpdir("#{@prefix.tr "@", "-"}-", HOMEBREW_TEMP))
```

But hold off until we decide whether we're going to merge this in any form, thanks!

Reply:
If change `dup.tr` into `tr`, then the original `@prefix` will be modified also, so homebrew will try to install at `$HOMEBREW_PREFIX/Cellar/gcc-10` (for example).

Reply:
Nope, `.tr!` will change the original `@prefix` but `.tr` will return a new string.

Reply:
Oh ok, thanks for the fix!